prse-derive/src/instructions.rs

```diff
@@ -220,39 +220,77 @@ impl Instructions {
+                                .map(|p| p.lending_parse());
+                        }
                     });
                 }
-                Instruction::MultiParse(_, sep, count) => {
+                Instruction::MultiParse(_, sep, count, is_multi) => {
                     let sep = string_to_tokens(sep);
                     let i = 0..*count;
-                    store_token = Some(quote! {
-                        let mut __prse_iter = __prse_parse.split(#sep)
-                            .map(|p| p.lending_parse());
-                        let #var = [ #(
-                            __prse_iter.next()
-                            .ok_or_else(|| ::prse::ParseError::Multi {
-                                expected: #count,
-                                found: #i,
-                            })??
-                        ),* ];
-                        if __prse_iter.next().is_some() {
-                            return Err(::prse::ParseError::Multi {
-                                expected: #count,
-                                found: #count + 1,
-                            });
+                    store_token = Some(if *is_multi {
+                        quote! {
+                            let mut __prse_iter = __prse_parse.split(#sep)
+                                .filter(|p|!p.is_empty())
+                                .map(|p| p.lending_parse());
+                            let #var = [ #(
+                                __prse_iter.next()
+                                .ok_or_else(|| ::prse::ParseError::Multi {
+                                    expected: #count,
+                                    found: #i,
+                                })??
+                            ),* ];
+                            if __prse_iter.next().is_some() {
+                                return Err(::prse::ParseError::Multi {
+                                    expected: #count,
+                                    found: #count + 1,
+                                });
+                            }
+                        }
+                    } else {
+                        quote! {
+                            let mut __prse_iter = __prse_parse.split(#sep)
+                                .map(|p| p.lending_parse());
+                            let #var = [ #(
+                                __prse_iter.next()
+                                .ok_or_else(|| ::prse::ParseError::Multi {
+                                    expected: #count,
+                                    found: #i,
+                                })??
+                            ),* ];
+                            if __prse_iter.next().is_some() {
+                                return Err(::prse::ParseError::Multi {
+                                    expected: #count,
+                                    found: #count + 1,
+                                });
+                            }
```

Comment:
I was planning on reworking the way iterators work (i.e. returning a concrete `parseIter` type instead of the current opaque types generated from `split` and `map` so **no need to change this** but in the future I would recommend doing something like this to prevent massive code duplication:
```rust
let iter = if *is_multi {
		quote!{__prse_parse.split(#sep)
                                .filter(|p|!p.is_empty())
                                .map(|p| p.lending_parse());}
	} else {
		quote!{__prse_parse.split(#sep)
                                .filter(|p|!p.is_empty());}
        };
```
So that you can just use the `iter` var and remove the massive if branch
```rust
let mut __prse_iter = #iter;
```

Reply:
:+1: Not a problem